icecreamshop/icecreamshop/src/main/java/com/summerschool/icecreamshop/service/ProductService.java

```diff
@@ -5,13 +5,22 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Map;
 import java.util.Optional;
 
 @Service
 public class ProductService {
     @Autowired
     ProductRepository productRepository;
 
+    public Product update(Product product) throws Exception {
+         if (productRepository.findById(product.getId()).isEmpty()) {
+            throw new Exception("404 Not Request");
+        } else {
+            return productRepository.save(product);
```

Comment:
please review the method that you are using and consider using 'update', the save method does not update an existing entity but only creates a new entry.

Maybe this will help: https://www.baeldung.com/hibernate-save-persist-update-merge-saveorupdate

Reply:
In articolul din link se utilizeaza obiectul session direct din hibernate. Intradevar aici metoda save face insert.
In cod se utilizeaza Jpa si anume interfata CrudRepository. Daca product are setat id metoda save face update. Daca nu este setat id face insert. Asa am inteles eu. E bine?